ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp

```diff
@@ -0,0 +1,414 @@
+// [WriteFile Name=ViewshedLocation, Category=Analysis]
+// [Legal]
+// Copyright 2017 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "ViewshedLocation.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "LocationViewshed.h"
+#include "Camera.h"
+#include "Point.h"
+#include "AnalysisOverlay.h"
+#include "Graphic.h"
+#include "SimpleMarkerSceneSymbol.h"
+#include "SimpleRenderer.h"
+
+using namespace Esri::ArcGISRuntime;
+
+ViewshedLocation::ViewshedLocation(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void ViewshedLocation::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<ViewshedLocation>("Esri.Samples", 1, 0, "ViewshedLocationSample");
+}
+
+void ViewshedLocation::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // Create a scene and give it to the SceneView
+  m_sceneView = findChild<SceneQuickView*>("sceneView");
+
+  Scene* scene = new Scene(Basemap::topographic(this), this);
+  Surface* surface = new Surface(this);
+  surface->elevationSources()->append(
+        new ArcGISTiledElevationSource(
+          QUrl("http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"),
+          this));
+  scene->setBaseSurface(surface);
+  m_sceneView->setArcGISScene(scene);
+
+  // Add an Analysis Overlay and a Graphics Overlay
+  m_analysisOverlay = new AnalysisOverlay(this);
+  m_sceneView->analysisOverlays()->append(m_analysisOverlay);
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  m_sceneView->graphicsOverlays()->append(m_graphicsOverlay);
+
+  // set initial viewpoint
+  setInitialViewpoint();
+
+  // connect signals
+  connectSignals();
+}
+
+void ViewshedLocation::setInitialViewpoint()
+{
+  // Set a viewpoint
+  double x = 6.86088;
+  double y = 45.3604;
+  double z = 3582.55;
+  Point point(x, y, z, SpatialReference(4326));
+  double heading = 345;
+  double pitch = 70;
+  double roll = 0;
+  Camera camera(point, heading, pitch, roll);
```

Comment:
Are these values stored in local variables for easier readability?

Reply:
Yes, we are trying to be more explicit in our samples. If you pass things into a function or constructor, and they take the same type for several arguments, it is ambiguous to a reader as to what the value is for, so they then would need to go to the doc to understand the code

Reply:
Make sense 👍